### PR TITLE
MEN-2760 report error on login when there is no netwok connection

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: src/js/api/users-api.js [hunk 1]
```diff
@@ -34,6 +36,14 @@ const Api = {
         .set('Content-Type', 'application/json')
         .send(body)
         .end((err, res) => {
+          if (err && !res) {
+            var errorResponse2 = {
+              text: err.response ? JSON.parse(err.response.text) : err,
+              code: err.status
+            };
+            AppActions.setSnackbar(preformatWithRequestID(err.res, err.message), null, 'Copy to clipboard');
```
Comment (mzedel) [inline src/js/api/users-api.js]: The api/ non-view layer of the application should be kept free from GUI side effects to keep them a bit more stable than the ever changing components/ view layer...

File: src/js/api/users-api.js [hunk 2]
```diff
@@ -34,6 +36,14 @@ const Api = {
         .set('Content-Type', 'application/json')
         .send(body)
         .end((err, res) => {
+          if (err && !res) {
+            var errorResponse2 = {
```
Comment (mzedel) [inline src/js/api/users-api.js]: that duplicates what happens in the other case `(err || !res.ok)` - is there any particular reason not to adjust the existing condition that I might be missing?

Comment (merlin-northern) [reply src/js/api/users-api.js]: > that duplicates what happens in the other case `(err || !res.ok)` - is there any particular reason not to adjust the existing condition that I might be missing?

I am sure you are not missing anything.

**Comment by merlin-northern** [comment]
> Thanks for the prompt contribution! Now that the sprint is over I finally got around to do a review, just to be complete... Since we have 2 PRs from around almost the same time I suggest closing this one and move forward with #538, but if you would like to see this contribution make it to production you will have to adjust your current implementation. In any case: I sure hope to see more of your excursions to the JS land in the future.

thanks!
feel free to close this one. I am sorry I did not know you already had the fix.

**Comment by mzedel** [comment]
closing for #538

**Pull request closed without merging**